src/guides/metamask-snap/index.md

```diff
@@ -0,0 +1,1324 @@
+      ), 
+    } }
+    disabled={false}
+    fullWidth={false}
+  />
+)}
+```
+
+To run through the entire life cycle you can take these steps:
+
+1. Import a funded account to your MetaMask wallet using the private keys generated by `ganache`
+2. Hit connect to install the snap
+3. Mint an NFT (you can put in an arbitrary string for the tokenURI). If this is the first mint, the token ID will be 1.
+4. Approve the NFTVault contract to `transfer` the NFT. Copy and paste the `SimpleNFT` address, which you can find in the `SimpleNFT.json` file or by calling `truffle networks`
+5. Deposit the NFT. For the second signer, use another pre-funded account number from the list of accounts generated by `ganache`
+6. Approve the withdrawal. Import the second signer account into your wallet by importing the private key. Switch and connect to that account to send the approval.
+7. Withdraw the NFT. You'll need to switch back to the original account in order to do the withdrawal.
+
+### Let's build the snap!
+
+This snap will do a few things:
+
+1. It will be able to identify which smart contract we are calling
+2. If we are calling the `NFTVault` contract, it will identify what method we are calling
+3. If we are calling `withdrawNFT`, it will communicate whether or not the NFT is approved to be withdraw and who the approver is
+
+### Edit `packages/snap/snap.manifest.json`
+
+We'll need to give new permissions to your snap. To do so, add `"endowment:transaction-insight": {}` to `initialPermissions`. This permission gives the snap read-only access to raw transaction payloads before they are accepted for signing by the user.
+
+### Edit `packages/snap/tsconfig.json`
+
+In order to access the contract build files, we'll need to add this to the `tsconfig.json` file:
+
+```javascript
+"compilerOptions": {
+  "resolveJsonModule": true
+}
+```
+
+### Edit `packages/snap/src/index.ts`
+
+Then, for our snap to use our transaction insight modify the imports:
+
+```javascript
+import {
+  OnTransactionHandler,
+  OnRpcRequestHandler,
+} from '@metamask/snap-types';
+import { getInsights } from './insights';
+```
+
+Then, add the following function:
+
+```javascript
+/**
+ * Handle an incoming transaction, and return any insights.
+ *
+ * @param args - The request handler args as object.
+ * @param args.transaction - The transaction object.
+ * @returns The transaction insights.
+ */
+export const onTransaction: OnTransactionHandler = async ({ transaction }) => {
+  return {
+    insights: await getInsights(transaction),
+  };
+};
+```
+
+### Create `packages/snap/src/insights.ts`
+
+First, we'll need to install the necessary dependencies:
+
+```shell
+cd ../snap
+yarn add @metamask/abi-utils ethers
+```
+
+Then create a new file called `insights.ts`. Add this code, and remember to replace `SimpleNFTContractAddress` and `NFTVaultContractAddress` with your migrated contracts. You can access those quickly by calling `truffle networks` from the `packages/truffle` folder.
```

Comment:
It looks like the need to replace `SimpleNFTContractAddress` and `NFTVaultContractAddress` is now redundant?